src/api/block_builder/input_selection/remainder.rs

```diff
@@ -22,6 +24,76 @@ use crate::{
     Error, Result,
 };
 
+// Get possible required storage deposit return outputs, if there is already an output for the storage deposit return,
+// then don't return a new output for that.
+pub(crate) fn get_storage_deposit_return_outputs<'a>(
+    inputs: impl Iterator<Item = &'a InputSigningData> + Clone,
+    outputs: impl Iterator<Item = &'a Output> + Clone,
+    current_time: u32,
+) -> Result<Vec<Output>> {
+    log::debug!("[get_storage_deposit_return_outputs]");
+
+    // Get inputs with storage deposit return unlock condition that are not expired
+    let inputs_sdr = inputs
+        .filter(|i| sdr_not_expired(&i.output, current_time).is_some())
+        // Safe to unwrap since we only return outputs with a sdr when filtering with sdr_not_expired()
+        .map(|i| {
```

Comment:
You can clean all this using https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.filter_map